src/GeositeFramework/plugins/layer_selector_v2/main.js

```diff
@@ -51,25 +51,96 @@ define([
                 this.filterTmpl = _.template(this.getTemplateByName('filter'));
                 this.treeTmpl = _.template(this.getTemplateByName('tree'));
                 this.layerTmpl = _.template(this.getTemplateByName('layer'));
+                this.infoBoxTmpl = _.template(this.getTemplateByName('info-box'));
+                this.layerMenuTmpl = _.template(this.getTemplateByName('layer-menu'));
                 this.bindEvents();
             },
 
             bindEvents: function() {
                 var self = this;
                 $(this.container)
-                    .on('click', 'a[data-layer-id]', function() {
+                    .on('click', 'a.layer-row', function() {
                         var $el = $(this),
-                            layerId = $el.attr('data-layer-id');
+                            $parent = $el.closest('li'),
+                            layerId = $parent.attr('data-layer-id');
                         self.state.toggleLayer(layerId);
                     })
-                    .on('keyup', '.pluginLayerSelector-search input', function() {
+                    .on('click', 'a.info', function() {
+                        var $el = $(this),
+                            $parent = $el.closest('li'),
+                            layerId = $parent.attr('data-layer-id');
+                        self.showLayerInfo(layerId);
+                    })
+                    .on('click', '.info-box .close', function() {
+                        self.hideLayerInfo();
+                    })
+                    .on('click', 'a.more', function() {
+                        var $el = $(this),
+                            $parent = $el.closest('li'),
+                            layerId = $parent.attr('data-layer-id'),
+                            $menu = self.createLayerMenu(layerId),
+                            $shadow = self.createLayerMenuShadow(),
+                            pos = $el.offset();
+                        $menu.css({
+                            top: pos.top,
+                            left: pos.left
+                        });
```

Comment:
We could also probably abstract out a function for getting the nearest `layerId`.